lib/orderbook/OrderBook.ts

```diff
@@ -1,130 +1,205 @@
 import uuidv1 from 'uuid/v1';
 
-import OrderBookRepository, { Orders } from './OrderBookRepository';
+import OrderBookRepository from './OrderBookRepository';
 import MatchingEngine from './MatchingEngine';
 import MatchesProcessor from './MatchesProcessor';
 import errors from './errors';
 import Pool from '../p2p/Pool';
 import { orders, matchingEngine, db } from '../types';
 import DB from '../db/DB';
-import { groupBy } from '../utils/utils';
 import Logger from '../Logger';
 import LndClient from '../lndclient/LndClient';
+import { StampedOrder, OutgoingOrder } from '../types/orders';
+import { ms } from '../utils/utils';
+import Peer from '../p2p/Peer';
+import OrdersPacket from '../p2p/packets/OrdersPacket';
 
 type OrderBookConfig = {
   internalmatching: boolean;
 };
 
+type Orders = {
+  buyOrders: StampedOrder[];
+  sellOrders: StampedOrder[];
+};
+
 class OrderBook {
+  public pairs: db.PairInstance[] = [];
+  public matchingEngines: { [ pairId: string ]: MatchingEngine } = {};
+
   private logger: Logger = Logger.orderbook;
   private repository: OrderBookRepository;
   private matchesProcessor: MatchesProcessor = new MatchesProcessor();
-  public pairs: db.PairInstance[] = [];
-  public matchingEngines: {[ pairId: string ]: MatchingEngine} = {};
+
+  private ownOrders: { [ pairId: string ]: Orders } = {};
+  private peerOrders: { [ pairId: string ]: Orders } = {};
 
   constructor(private config: OrderBookConfig, db: DB, private pool?: Pool, private lndClient?: LndClient) {
     this.repository = new OrderBookRepository(db);
     if (pool) {
       pool.on('packet.order', this.addPeerOrder);
+      pool.on('packet.getOrders', async (peer: Peer) => {
+        // TODO: just send supported pairs
+        const pairs = await this.getPairs();
+        const outgoingOrders: OutgoingOrder[] = [];
+
+        for (const { id } of pairs) {
+          const orders = await this.getOwnOrders(id, 0);
+          await Promise.all([
+            this.castToOutgoingOrder(orders.buyOrders, outgoingOrders),
+            this.castToOutgoingOrder(orders.sellOrders, outgoingOrders),
+          ]);
+        }
+
+        const packet = new OrdersPacket({ orders: outgoingOrders });
```

Comment:
`OrderBook` shouldn't construct a new packet. You already implement `sendOrders` under `Peer`. You should just pass it `outgoingOrders`.